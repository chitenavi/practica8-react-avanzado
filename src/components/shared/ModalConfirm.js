import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';
import './ModalConfirm.scss';

const ModalConfirm = ({ children, title, show, onClose }) => {
  return show ? (
    <div className="modalConfirm">
      <div className="modalConfirm-dialog">
        <h2 className="modalConfirm-dialog--title">{title}</h2>
        <div className="modalConfirm-dialog--content">{children}</div>
        <div className="modalConfirm-dialog--actions">
          <Button className="secondary" onClick={() => onClose(false)}>
            Cancel
          </Button>
          <Button className="tertiary" onClick={() => onClose(true)}>
            Confirm
          </Button>
        </div>
      </div>
    </div>
  ) : null;
};

ModalConfirm.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  show: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
};
export default ModalConfirm;
