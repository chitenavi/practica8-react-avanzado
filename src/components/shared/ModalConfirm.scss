@import '../../scss/variables';

.modalConfirm {
  background-color: rgba(255, 255, 255, 0.6);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;

  &-dialog {
    max-width: 400px;
    min-height: 150px;
    background-color: $bg-color-sec;
    padding: 0.5rem 2rem;
    border-radius: 0.5rem;
    display: flex;
    flex-direction: column;
    justify-content: space-around;

    &--title {
      text-align: center;
      text-transform: uppercase;
      color: white;
    }
    &--content {
      color: white;
      font-size: 1.1rem;
      text-align: center;
    }

    &--actions button {
      margin: 0 0.5rem;
    }
  }
}
