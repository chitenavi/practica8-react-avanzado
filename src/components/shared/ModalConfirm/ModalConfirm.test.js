import React from 'react';
import { shallow } from 'enzyme';

import ModalConfirm from './ModalConfirm';

describe('ModalConfirm', () => {
  const children = 'something';
  const show = false;
  const title = 'Modal';
  const onClose = jest.fn();

  const render = () =>
    shallow(
      <ModalConfirm show={show} title={title} onClose={onClose}>
        {children}
      </ModalConfirm>,
    );

  const wrapper = render();

  it('should not render component if show prop is false', () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.isEmptyRender()).toBe(true);
  });

  it('should render component when show prop is true', () => {
    wrapper.setProps({ show: true });
    expect(wrapper.exists()).toBe(true);

    expect(wrapper.find('.modalConfirm-dialog--title').text()).toEqual(title);
    expect(wrapper.find('.modalConfirm-dialog--content').text()).toEqual(
      children,
    );
  });

  it('should call onClose function', () => {
    const button = wrapper.find('.secondary');

    button.simulate('click');

    expect(onClose).toHaveBeenCalled();
  });

  it('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
