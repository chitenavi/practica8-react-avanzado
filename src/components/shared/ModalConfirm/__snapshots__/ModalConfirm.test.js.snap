// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalConfirm snapshot 1`] = `
<div
  className="modalConfirm"
>
  <div
    className="modalConfirm-dialog"
  >
    <h2
      className="modalConfirm-dialog--title"
    >
      Modal
    </h2>
    <div
      className="modalConfirm-dialog--content"
    >
      something
    </div>
    <div
      className="modalConfirm-dialog--actions"
    >
      <Button
        className="secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Cancel
      </Button>
      <Button
        className="tertiary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Confirm
      </Button>
    </div>
  </div>
</div>
`;
